src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/RazorDiagnosticsEndpoint.cs

```diff
@@ -166,45 +165,67 @@ public async Task<RazorDiagnosticsResponse> Handle(RazorDiagnosticsParams reques
 
             var filteredDiagnostics = unmappedDiagnostics
                 .Where(d =>
-                    // TODO: undesired HTML Diagnostic codes
-                    // (https://dev.azure.com/devdiv/DevDiv/_workitems/edit/1257401) ||
-                    !InAttributeContainingCSharp(d, sourceText, syntaxTree, processedAttributes))
+                    !InAttributeContainingCSharp(d, sourceText, syntaxTree, processedAttributes) &&
+                    !FilterDiagnosticBasedOnErrorCode(d, sourceText, syntaxTree))
                 .ToArray();
 
             return filteredDiagnostics;
+        }
+
+        private static bool FilterDiagnosticBasedOnErrorCode(Diagnostic d, SourceText sourceText, RazorSyntaxTree syntaxTree)
+        {
+            if (!d.Code.HasValue)
+            {
+                return false;
+            }
+
+            return d.Code.Value.String switch
+            {
+                HtmlErrorCodes.InvalidNestingErrorCode => IsInvalidNestingWarningWithinComponent(d, sourceText, syntaxTree),
+                HtmlErrorCodes.MissingEndTagErrorCode => true, // Redundant with RZ9980
```

Comment:
So we can't always suppress this diagnostic. In Blazor scenarios "missing end tag" equates to an error; however, in non-Blazor scenarios it's allowed.

Reply:
Ah good catch, thanks!

I believe https://github.com/dotnet/aspnetcore-tooling/pull/3208/commits/d7a96bd142a6df844bcc95adac3f00a1f564f408 should address this.

![image](https://user-images.githubusercontent.com/14852843/109112268-3433bf80-76ef-11eb-80fa-7e8f6eaf267f.png)